bugs/gh-267.txt

```diff
@@ -0,0 +1,795 @@
+=== Begin SRU Template ===
+[Impact]
+Ensure ENI network rendering is preferred ahead of netplan on
+Xenial, Bionic and Eoan even if both ifupdown and netplan are installed.
+
+Focal and later will prioritize netplan ahead of ENI even if both ifupdown and
+netplan are present.
+
+[Test Case]
+  1. launch cloudimage with current cloud-init using cloud-init to installl
+     whatever package is missing: ifupdown or nplan/netplan.io.
+     (nplan was the pkg name in xenial, netplan.io in bionic and later)
+  2. cloud-init clean --reboot --logs 
+  3. Validate the default renderer chosen (eni on all series; which is broken on bionic and later)
+  4. Upgrade cloud-init to 20.2.45 from -proposed; cloud-init clean reboot
+  5. Validate correct default renderer chosen: 
+      - eni remains chosen on xenial, bionic no priority list set
+      - bionic/eoan breaks because of no fix for LP:1867029 and no priority
```

Comment:
Can you explain this line? Isn't LP:1857029 what we're testing here? Are you saying that PR only fixed this issue for focal?

Reply:
@TheRealFalcon correct. We only actually released a fix for LP:1857029 for focal. Any release before then will exhibit an issue with cloud-inint continuing to render  /etc/network/interfaces/50-cloud-init.cfg, but in default cloud images, the file /etc/network/interfaces does not properly have a line "source /etc/network/interfaces.d/*" so that rendered network config by ifupdown doesn't actually see the ENI/50-cloud-init.cfg